cogito/protocol.go

```diff
@@ -0,0 +1,184 @@
+// This file implements the Concourse resource protocol described at
+// https://concourse-ci.org/implementing-resource-types.html
```

Comment:
Concourse passes to the stdin of the check, in, out executables a partially different JSON object.

This file parses, validates and sets default values both of the stdinput and of other metadata such as environment variables.